tests/src/Kernel/ConfigEntityTest.php

```diff
@@ -66,6 +76,46 @@ public function testConfigAction() {
   }
 
   /**
+   * Test saving a deeply nested rule.
+   *
+   * @todo This test currently fails if $strictConfigSchema = TRUE.
+   */
+  public function testNestedRuleSave() {
+    // This test is specific to reaction rules, so get the appropriate
+    // entity storage:
+    $storage = $this->container->get('entity.manager')->getStorage('rules_reaction_rule');
+    // Create a nested expression.
+    $and = $this->expressionManager->createAnd()
+      ->addCondition('rules_test_false')
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->negate();
+
+    $or = $this->expressionManager->createOr()
+      ->addCondition('rules_test_true', ContextConfig::create()->negateResult())
+      ->addCondition('rules_test_false')
+      ->addCondition($and);
```

Comment:
Changing this does make the test pass w/o the fix code.

So it was an error, but a subtle error.  We probably should enforce is_string() on createAction() and createCondition(), or at least use type hinting to get PHP to put out a warning.  This way, the problem is revealed at the source, rather than only when the config entity is saved.